extensions/mapstore/src/main/java/com/hazelcast/mapstore/MappingHelper.java

```diff
@@ -82,6 +84,47 @@ private String createMappingQuery(
         DIALECT.quoteStringLiteral(sb, null, idColumn);
         sb.append(" )");
         return sb.toString();
+
+    }
+
+    //package-private just for testing
+    static String externalName(String tableName) {
+        return splitByNonQuotedDots(tableName).stream()
+                .map(unquoteIfQuoted())
+                .map(DIALECT::quoteIdentifier)
+                .collect(Collectors.joining("."));
```

Comment:
I think we should keep one style for the query-building parts - I tried to settle on using an imperative style with StringBuilder.

Reply:
nit : The method name can be quoteExternalName

Reply:
Refactored to StringBuilder although I find it more error prone https://github.com/hazelcast/hazelcast/pull/24332/commits/3071b66ad89d7bda39fb4219536e2729924a71be

Renamed https://github.com/hazelcast/hazelcast/pull/24332/commits/cb74c881e6a1affb69e60e736fa11ea70d1a6003